### ISSUE TITLE
AWS::RDS::DBCluster ServerlessV2

### ISSUE THREAD

**Issue by MrSakhs** [aws-cloudformation/cloudformation-coverage-roadmap#1153]
### Name of the resource

AWS::RDS::DBCluster

### Description

[Amazon Aurora Serverless v2 is generally available ](https://aws.amazon.com/about-aws/whats-new/2022/04/amazon-aurora-serverless-v2/) was announced on the 21st of April (a week ago). This is feature request for CloudFormation to cover the gap and support Aurora Serverless V2 on the resource `AWS::RDS:DBCluster`, just like serverless version 1 is supported.
 
Example for Serverless Version 1 found here "[Creating an Amazon Aurora Serverless DB cluster](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#aws-resource-rds-dbcluster--examples--Creating_an_Amazon_Aurora_Serverless_DB_cluster) "

### Other Details

I am not clear if the [issue 1150](https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/1150) is covering this, but the focus is on `AWS::RDS::DBInstance`. Please check on this.

**Comment by robertrossmann** [comment]
For context, the main issue preventing us from deploying Serverless v2 with CloudFormation is the lack of support for the [`ServerlessV2ScalingConfiguration`](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ServerlessV2ScalingConfiguration.html) configuration property on the cluster. This property must be present on an AWS::RDS::DBCluster resource before we can deploy a `db.serverless` instance into it, otherwise the deployment fails due to missing configuration.

**Comment by iwanbolzern** [comment]
+1

**Comment by cmesas** [comment]
+1

**Comment by rsung-deepwatch** [comment]
+1

**Comment by Lettly** [comment]
any news?

**Comment by huwd** [comment]
+1 :)

**Comment by siebrand** [comment]
Please use thumbs up on the first post. "+1" or "any news" are discouraged. From the repo readme:

```
Q: Can I "+1" existing issues?

A:We strongly encourage you to do so, as it helps us understand which issues will have the widest impact. You can navigate to the issue details page and add a reaction (thumbs up). There are six types of reactions supported (thumbs down “-1”, confused, heart, watching, laugh, hooray, and thumbs up +1). We can typically do 8-10 items per month with all necessary regression testing, and we want you to help us decide which items will benefit you the most.
```

**Comment by shanemacphillamy** [comment]
+1

**Comment by VanDuursenTopicus** [comment]
+1

**Comment by ClaudenirFreitas** [comment]
+1

**Comment by snakelab** [comment]
+1

**Comment by Beagle** [comment]
+1

**Comment by vhermecz** [comment]
First, thx for the example link in the last comment @MrSakhs, that gave it finally away.
While this is not entirely impossible indeed, the current implementation feels way too fragile.

TLDR:
- If you want v1 serverless, you need to set `EngineMode: serverless` on the `AWS::RDS::DBCluster`.
- If you want v2 serverless, you need to set `DBInstanceClass: db.serverless` on a `AWS::RDS::DBInstance` referencing the cluster, and DO NOT SET `EngineMode: serverless`
- Having `EngineMode: serverless` locks you in into v1 serverless. As if v2 would not be serverless. **If you want v2 serverless, you should not use engine mode serverless.**

Thoughts:
- If both are called serverless, and they only differ in a version, wouldn't it be a better abstraction to have a `ServerlessVersion` property?
- Or couldn't at least CFN take the presence of the `ServerlessV2ScalingConfiguration` property as a hint for v2?
- Also simply using a `DBClusterInstanceClass` property on the `DBCluster` gives an error. This field sounds like a shorthand for creating a separate `AWS::RDS::DBInstance`, but it is not behaving that way when it comes to serverless :/

```
Resources:
  ...
  DatabaseInstance1:
    Type: AWS::RDS::DBInstance
    Properties:
      DBClusterIdentifier: !Ref RDSCluster
      Engine: aurora-postgresql
      DBInstanceClass: db.serverless  # Need this to signal Serverless v2
  RDSCluster:
    Type: AWS::RDS::DBCluster
    Properties:
      DBClusterIdentifier: !Ref DBClusterName
      # DBClusterInstanceClass: db.serverless  # DONT SET! Needs a separate 'AWS::RDS::DBInstance' to set it
      MasterUsername: !Join ['', ['{{resolve:secretsmanager:', !Ref DBSecret, ':SecretString:username}}' ]]
      MasterUserPassword: !Join ['', ['{{resolve:secretsmanager:', !Ref DBSecret, ':SecretString:password}}' ]]
      DatabaseName: !Ref DatabaseName
      Engine: aurora-postgresql
      # EngineMode: serverless  # DONT SET! Enforces Serverless v1
      EngineVersion: 13.6
      ServerlessV2ScalingConfiguration:
        MinCapacity: 0.5
        MaxCapacity: 2
      DBSubnetGroupName:
        Ref: DBSubnetGroup
```
